js/csv.js

```diff
@@ -0,0 +1,227 @@
+/**
+ * Reads a CSV file and returns a table with the data.
+ *
+ * @param {Blob|File} file The file object to be read.
+ * @param {Function} parser The function to be used in the
+ * CSV parsing operation.
+ */
+export const parseCsvFile = (file, parser = null) => {
+    parser = parser || parseCsvComplex;
+    return new Promise((resolve, reject) => {
+        const reader = new FileReader();
+        reader.onload = e => resolve(parser(e.target.result));
+        reader.onerror = e => reject(e);
+        reader.readAsText(file);
+    });
+};
+
+/**
+ * Creates the contents of the CSV file with the data given
+ * as an array of rows and the headers.
+ *
+ * @param {Array / Object} data Data to be converted into CSV. Can be
+ * an array of strings, a JSON object, or an array of those.
+ * @param {Array} headers List of headers for the CSV file, is an array of strings.
+ * @param {String} delimiter The delimiter to be used in the CSV.
+ * @returns A string containing the information for the CSV.
+ */
+export const buildCsv = (data, headers = [], delimiter = ",") => {
+    let csv = "";
+    if (Object.keys(headers).length !== 0) csv += headers.join(delimiter) + "\n";
+
+    // is simple object
+    if (!Array.isArray(data)) {
+        return csv + _objectToCsv(data, headers, delimiter);
+    }
+
+    // is simple array
+    if (typeof data[0] !== "object") return csv + _arrayToCsv(data, delimiter);
+    if (Object.keys(data).length === 0) return "";
+
+    for (let i = 0; i < Object.keys(data).length; i++) {
+        // check if array, if it is add value as is
+        if (Array.isArray(data[i])) {
+            csv += _arrayToCsv(data[i], delimiter);
+            continue;
+        }
+
+        csv += _objectToCsv(data[i], headers, delimiter);
+    }
+
+    return csv;
+};
+
+/**
+ * Converts an object into a CSV.
+ *
+ * @param {Object} data The object to be converted.
+ * @param {Array} headers The headers that are going to be
+ * searched for. If not existent, use keys for current
+ * data being converted.
+ * @param {String} delimiter The delimiter for the CSV.
+ * @returns A string of the converted object.
+ */
+export const _objectToCsv = (data, headers, delimiter = ",") => {

```

Comment:
It was a way to remove duplication of logic, since it was used in both [here](https://github.com/ripe-tech/ripe-commons-js/blob/d9bdb499cd23e0a311e7bbf1c9c62cb4ec3cf8d2/js/csv.js#L34) and [here](https://github.com/ripe-tech/ripe-commons-js/blob/d9bdb499cd23e0a311e7bbf1c9c62cb4ec3cf8d2/js/csv.js#L48)

Reply:
I'm asking to know more about the why there is this logic to convert an object to a CSV line? Where did the requirement came from?

Looking at the original mixin, the `buildCsv()` was way simpler

Reply:
That's because the original `buildCsv()` couldn't build a CSV out of an object. The original CSV was simpler, but this is much more feature complete

Reply:
But was it a requirement or you just added because you thought it would improve the `buildCsv()`?
I ask because if it's the latter, I propose removing this `_objectToCsv()` logic as is too much error prone.

Just keep it simple, if someone has an object and wants to use the `buildCsv()` let them handle the parsing of the data by themselves and call the `buildCsv()` as they need. The `buildCsv()` method is way complex and assuming a lot thus very error prone